app/scripts/metamask-controller.js

```diff
@@ -1514,10 +1513,32 @@ export default class MetamaskController extends EventEmitter {
       signatureController: this.signatureController,
       platform: this.platform,
       extension: this.extension,
+      getTransactions: this.txController.getTransactions.bind(
+        this.txController,
+      ),
+      setTxStatusSigned:
+        this.txController.txStateManager.setTxStatusSigned.bind(
+          this.txController.txStateManager,
+        ),
+      setTxStatusSubmitted:
+        this.txController.txStateManager.setTxStatusSubmitted.bind(
+          this.txController.txStateManager,
+        ),
+      setTxStatusFailed:
+        this.txController.txStateManager.setTxStatusFailed.bind(
+          this.txController.txStateManager,
+        ),
       trackTransactionEvents: handleMMITransactionUpdate.bind(
         null,
         transactionMetricsRequest,
       ),
+      updateTransaction:
+        this.txController.txStateManager.updateTransaction.bind(
+          this.txController.txStateManager,
+        ),
+      updateTransactionHash: this.txController.setTxHash.bind(
```

Comment:
LGTM! 🥇 
Just out of curiosity why not keeping the same name `setTxHash` and have the new `updateTransactionHash`? Will the method itself in `txController` also get the same name? I actually find `updateTransactionHash` more easy to understand so good change 👍🏼

Reply:
We are including a method `updateCustodialTransaction` to support the existing use case.